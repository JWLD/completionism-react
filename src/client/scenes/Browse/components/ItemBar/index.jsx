import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { ICON_URLS } from 'constants/urls'
import { setActiveItemId } from 'Browse/actions'
import { activeCategorySelector } from 'Browse/selectors'

import * as SC from 'ItemBar/styled'

export const ItemBar = ({
  category,
  collected,
  icon,
  id,
  level,
  name,
  quality,
  setActiveItemId
}) => {
  const iconUrl = icon ? `url(${ICON_URLS.large}${icon}.jpg)` : 'none'

  let progressIcon = null

  if (collected && category === 'pets') {
    progressIcon = <SC.PetLevel level={level}>{level}</SC.PetLevel>
  } else if (collected) {
    progressIcon = <SC.CheckIcon />
  } else {
    progressIcon = <SC.CrossIcon />
  }

  return (
    <SC.ItemBar onClick={() => setActiveItemId(id)}>
      <SC.ItemIcon iconUrl={iconUrl} />
      <SC.ItemTitle quality={quality}>{name}</SC.ItemTitle>
      {progressIcon}
    </SC.ItemBar>
  )
}

ItemBar.propTypes = {
  category: PropTypes.string.isRequired,
  collected: PropTypes.bool.isRequired,
  icon: PropTypes.string,
  id: PropTypes.number.isRequired,
  level: PropTypes.number,
  name: PropTypes.string.isRequired,
  quality: PropTypes.number.isRequired,
  setActiveItemId: PropTypes.func.isRequired
}

ItemBar.defaultProps = {
  icon: null,
  level: 0
}

const mapStateToProps = state => ({
  category: activeCategorySelector(state)
})

const mapDispatchToProps = {
  setActiveItemId
}

export default connect(mapStateToProps, mapDispatchToProps)(ItemBar)
