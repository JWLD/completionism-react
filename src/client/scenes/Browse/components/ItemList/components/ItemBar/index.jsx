import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { ICON_URLS } from 'constants/urls'
import { setActiveItem } from 'ItemList/actions'
import { activeCategorySelector } from 'Browse/selectors'

import * as SC from 'ItemBar/styled'

export const ItemBar = ({
  category,
  collected,
  icon,
  id,
  level,
  name,
  quality,
  setActiveItem
}) => {
  const iconUrl = icon ? `url(${ICON_URLS.large}${icon}.jpg)` : 'none'

  const renderProgressIcon = () => {
    if (collected && category === 'pets') {
      const maxPet = level === 0 || (quality === 3 && level === 25)
      const petLevel = level === 0 ? '-' : level

      return <SC.PetLevel maxPet={maxPet}>{petLevel}</SC.PetLevel>
    } else if (collected) {
      return <SC.CheckIcon />
    } else {
      return <SC.CrossIcon />
    }
  }

  return (
    <SC.ItemBar onClick={() => setActiveItem(id)}>
      <SC.ItemIcon iconUrl={iconUrl} />
      <SC.ItemTitle quality={quality}>{name}</SC.ItemTitle>
      {renderProgressIcon()}
    </SC.ItemBar>
  )
}

ItemBar.propTypes = {
  category: PropTypes.string.isRequired,
  collected: PropTypes.bool.isRequired,
  icon: PropTypes.string,
  id: PropTypes.number.isRequired,
  level: PropTypes.number,
  name: PropTypes.string.isRequired,
  quality: PropTypes.number.isRequired,
  setActiveItem: PropTypes.func.isRequired
}

ItemBar.defaultProps = {
  icon: null,
  level: 0
}

const mapStateToProps = state => ({
  category: activeCategorySelector(state)
})

const mapDispatchToProps = {
  setActiveItem
}

export default connect(mapStateToProps, mapDispatchToProps)(ItemBar)
