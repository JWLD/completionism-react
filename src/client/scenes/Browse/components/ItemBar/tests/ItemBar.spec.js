import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { ItemBar } from 'ItemBar'

Enzyme.configure({ adapter: new Adapter() })

let mockProps = {}

beforeEach(() => {
  mockProps = {
    collected: true,
    id: 1,
    name: 'foo',
    quality: 1,
    setActiveItemId: jest.fn()
  }
})

describe('<ItemBar />', () => {
  it('renders correctly', () => {
    const component = shallow(<ItemBar {...mockProps} />)

    expect(component).toMatchSnapshot()
  })

  it('renders when [collected] prop is false', () => {
    mockProps.collected = false
    const component = shallow(<ItemBar {...mockProps} />)

    expect(component).toMatchSnapshot()
  })

  it('renders when passed an [icon] prop', () => {
    mockProps.icon = 'foo'
    const component = shallow(<ItemBar {...mockProps} />)

    expect(component).toMatchSnapshot()
  })
})
