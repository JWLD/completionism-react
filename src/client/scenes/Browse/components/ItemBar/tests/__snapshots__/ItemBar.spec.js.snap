// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemBar /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemBar
    collected={true}
    icon={null}
    id={1}
    level={0}
    name="foo"
    quality={1}
    setActiveItemId={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(styled.i)
          iconUrl="none"
        />,
        <ItemTitle
          quality={1}
        >
          foo
        </ItemTitle>,
        <CheckIcon />,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconUrl": "none",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "foo",
          "quality": 1,
        },
        "ref": null,
        "rendered": "foo",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(styled.i)
            iconUrl="none"
          />,
          <ItemTitle
            quality={1}
          >
            foo
          </ItemTitle>,
          <CheckIcon />,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "iconUrl": "none",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "foo",
            "quality": 1,
          },
          "ref": null,
          "rendered": "foo",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ItemBar /> renders when [collected] prop is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemBar
    collected={false}
    icon={null}
    id={1}
    level={0}
    name="foo"
    quality={1}
    setActiveItemId={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(styled.i)
          iconUrl="none"
        />,
        <ItemTitle
          quality={1}
        >
          foo
        </ItemTitle>,
        <CrossIcon />,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconUrl": "none",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "foo",
          "quality": 1,
        },
        "ref": null,
        "rendered": "foo",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(styled.i)
            iconUrl="none"
          />,
          <ItemTitle
            quality={1}
          >
            foo
          </ItemTitle>,
          <CrossIcon />,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "iconUrl": "none",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "foo",
            "quality": 1,
          },
          "ref": null,
          "rendered": "foo",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ItemBar /> renders when passed an [icon] prop 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemBar
    collected={true}
    icon="foo"
    id={1}
    level={0}
    name="foo"
    quality={1}
    setActiveItemId={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(styled.i)
          iconUrl="url(http://wow.zamimg.com/images/wow/icons/large/foo.jpg)"
        />,
        <ItemTitle
          quality={1}
        >
          foo
        </ItemTitle>,
        <CheckIcon />,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "iconUrl": "url(http://wow.zamimg.com/images/wow/icons/large/foo.jpg)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "foo",
          "quality": 1,
        },
        "ref": null,
        "rendered": "foo",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(styled.i)
            iconUrl="url(http://wow.zamimg.com/images/wow/icons/large/foo.jpg)"
          />,
          <ItemTitle
            quality={1}
          >
            foo
          </ItemTitle>,
          <CheckIcon />,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "iconUrl": "url(http://wow.zamimg.com/images/wow/icons/large/foo.jpg)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "foo",
            "quality": 1,
          },
          "ref": null,
          "rendered": "foo",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
